File: C/other/Palindrome.c
/*
C Program to check whether a number is Palindrome or not
An integer is a palindrome if the reverse of that number is equal to the original number.
Example of Palindromes are 11,44,101,121,
*/
#include <stdio.h>

int reverse_num(int);
int main()
{
	int n, rev;
	printf("\n Enter a number:");
	scanf("%d", &n);
	rev = reverse_num(n);
	if (rev == n)
	{
		printf("\n %d is a palindrome", n);
	}
	else
	{
		printf("\n %d is not a palindrome", n);
	}
	return 0;
}
int reverse_num(int num)
{
	int rev = 0;
	int rem;
	while (num > 0)
	{
		rem = num % 10;
		rev = rev *10 + rem;
		num = num / 10;
	}
	return rev;
}

/*
Sample Output
Enter a number: 121
121 is a palindrome

Complexities
Time Complexity: O(n)
Space Complexity: O(1)
*/

File: Python/other/Palindrome.py
/*
Python Program to check whether a number is Palindrome or not
An integer is a palindrome if the reverse of that number is equal to the original number.
Example of Palindromes are 11,44,101,121.
*/

def palindrome(n):
    a = n
    sum = 0
    r = 0
    while n > 0:
        r = n % 10
        sum = sum * 10 + r
        n = n // 10
    if sum == a:
        print (a, 'is a palindrome')
    else:
        print (a, 'is not a palindrome')
n = eval(input('Enter No '))
palindrome(n)

/*
Sample Case:
Example 1:
Enter a number: 121
121 is a palindrome

Example 2:
Enter a number: 123
123 is not a palindrome

Time Complexity: O(n)
Space Complexity: O(1)
*/
